pkg/base/config.go

```diff
@@ -252,6 +252,17 @@ var (
 	// will send to a given follower without hearing a response.
 	defaultRaftMaxInflightMsgs = envutil.EnvOrDefaultInt(
 		"COCKROACH_RAFT_MAX_INFLIGHT_MSGS", 128)
+
+	// defaultRaftMaxInflightBytes specifies the maximum aggregate byte size of
+	// Raft log entries that a leader will send to a follower without hearing
+	// responses.
+	//
+	// TODO(#90314): lower this limit to something close to max rates observed in
+	// healthy clusters. Currently this value is a conservatively large multiple
+	// of defaultRaftMaxSizePerMsg * defaultRaftMaxInflightMsgs, so that it
+	// doesn't cut off traffic until we know it's ok.
```

Comment:
I wonder if the default here should not be hard-coded, but constructed dynamically based on MaxInflightMsgs and MaxMsgSize. Something like:

```go
maxInflightBytes := 256 << 20
if other := maxBytesPerMsg * maxInflightMsgs {
  maxInflightBytes = other
}
```

We have some customers that configure MaxInflightMsgs to very high values. Presumably they'd still end up below 256<<20 (in actual BDP) for the maxInflightBytes but it does make some sense for the maxInflightBytes to always be larger than the bandwidth-delay product implied by the other two settings, unless it's explicitly specified.

Reply:
Added a fixup commit (not merging yet, testing the idea). Does this look any reasonable?

Reply:
The idea is: if either `RaftConfig` or env variable has this option non-zero, then the user knows what they are doing, so we don't check. Otherwise we compute the default which is a multiple of the other 2 parameters (which can either be default too, or overridden by the user).

Alternatively, we could by default set `MaxInflightBytes` to 0, which would disable this feature. But then we would need to explicitly modify our prod environment to enable it. Which of the ways is better?

Reply:
Ended up using your suggestion.

I think `RaftMaxInflightBytes` is an important limit on its own, it shouldn't heavily depend on `RaftMaxBytesPerMsg` because the latter is too much of a soft limit. So I reverted my idea of dynamically setting it to `maxBytesPerMsg * maxInflightMsgs * 64` because that presumes that the largest message shouldn't normally exceed 64x target message size. I think that the users concerned with the bandwidth-delay product should primarily be tweaking `RaftMaxInflightBytes`.

Upping `RaftMaxInflightBytes` to the product makes sense, to not accidentally break users who already assume things based on the other 2 settings. Alternatively, we could panic if the config is not reasonable? But I don't know if that would break any existing users.